### PR TITLE
FIX : prevent error with realpath

### PR DESCRIPTION
**Pull request by defrance**
if some bad caracters are present in the name of file ( ° by exemple)

### CODE REVIEW

**Comment by eldy** [comment]
On which os do you experience pb with realpath with bad char ?
Can you try
realpath(dol_osencode($file)) ?

**Comment by defrance** [comment]
it's happend on windows OS, seem good on linux
with dol_osencode it's ok
I add dol_osencode, but leave the control in case other cases would not be accepted

File: htdocs/core/lib/images.lib.php [hunk 1]
```diff
@@ -77,7 +77,10 @@ function dol_getImageSize($file, $url = false)
 	$fichier = $file;
 	if (!$url)
 	{
-		$fichier = realpath($file); 	// Chemin canonique absolu de l'image
+		$fichier = realpath(dol_osencode($file)); 	// Chemin canonique absolu de l'image
```
Comment (eldy) [inline htdocs/core/lib/images.lib.php:80]: If you need dol_osencode of $file, you probably need it 3 lines later 
$dir = dirname($file);
don't you ?

**Comment by eldy** [comment]
Does this occurs on "all images" files, or only if you have a file name of path with a special char ?

**Comment by defrance** [comment]
in fact $dir is not used in the function, so i suppress the line
they occur with specific car, but with osencode they are no problem
the control is present to prevent some ignored case instead of

**Pull request merged**